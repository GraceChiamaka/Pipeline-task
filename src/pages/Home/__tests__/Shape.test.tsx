import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Shape from "../Shape";
import { ShapeProps } from "../Shape";

test("Should Render Shape Component", () => {
	render(<Shape shape="square" color="gray" />);
	const alertElement = screen.getByTestId("shape-display");
	expect(alertElement).toBeInTheDocument();
});

test("matches snapshot", () => {
	const shape: ShapeProps = {
		shape: "oval",
		color: "blue",
	};
	const tree = renderer
		.create(<Shape shape={shape.shape} color={shape.color} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
