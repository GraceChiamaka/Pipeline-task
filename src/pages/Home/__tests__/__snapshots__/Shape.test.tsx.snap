// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="shade-display-item"
  data-testid="shape-display"
>
  <div
    className="color-well oval blue"
  />
</div>
`;
